components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -256,16 +256,16 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
     ): Promise<void> {
         await super.mayStartWorkspace(ctx, user, runningInstances);
 
-        let result;
+        let result: MayStartWorkspaceResult = {};
         try {
             result = await this.entitlementService.mayStartWorkspace(user, new Date(), runningInstances);
-        } catch (error) {
-            throw new ResponseError(ErrorCodes.INTERNAL_SERVER_ERROR, `Error in Entitlement Service.`);
-        }
-        log.info("mayStartWorkspace", { result });
-        if (result.mayStart) {
-            return; // green light from entitlement service
+            TraceContext.addNestedTags(ctx, { mayStartWorkspace: { result } });
+        } catch (err) {
+            log.error({ userId: user.id }, "EntitlementSerivce.mayStartWorkspace error", err);
+            TraceContext.setError(ctx, err);
+            return; // we don't want to block workspace starts because of internal errors
```

Comment:
this means, all negative conditions should produce a negative response. if anything unexpected happens, then no rules are enforced. is that the intention? 

what's the expected feedback loop for us?

Reply:
The feedback loop should never involve the user. The error logged here should a) feed a log-rate metric, and b) we should have specific metrics for possible causes of this, e.g. the Stripe API.